cocina_level2_prs.rb

```diff
@@ -0,0 +1,185 @@
+def create_pr
+  ErrorEmittingExecutor.execute("hub pull-request -f -m '#{COMMIT_DESCRIPTION}'")
+end
+
+# Check the result string of ErrorEmittingExecutor command for failure,
+#   output a message if so
+def result_failure?(step_name, result)
+  if result == 'FAILED'
+    p "stopped after #{step_name} due to failure"
+    return true
+  end
+  false
+end
+
+def clone_or_update_repo(repo)
+  puts "\n**** Processing #{repo} ****"
+  if File.exist?(cloned_repo_dir(repo))
+    update_repo(repo)
+  else
+    create_repo(repo)
+  end
+end
+
+def update_repo(repo)
+  within_cloned_repo_dir(repo) do
+    fetch_update
+    # we need local main branch to be up to date if we are creating a branch based on it
+    ErrorEmittingExecutor.execute('git switch main', exit_on_error: true)
+    # This allows our default branches to vary across projects
+    ErrorEmittingExecutor.execute('git reset --hard $(git symbolic-ref refs/remotes/origin/HEAD)',
+                                  exit_on_error: true)
+    ErrorEmittingExecutor.execute('bundle install', exit_on_error: true)
+  end
+end
+
+# fetch main and our latest branch from origin if it exists
+# must be called as part of the block for within_cloned_repo_dir(repo)
+def fetch_update
+  remote_branch = ErrorEmittingExecutor.execute("git branch -r --list origin/#{BRANCH_NAME}")
+  if remote_branch == 'origin/cocina-level2-updates'
+    ErrorEmittingExecutor.execute("git fetch origin #{GIT_MAIN_FETCH_REFS} #{GIT_BRANCH_FETCH_REFS}")
+  else
+    ErrorEmittingExecutor.execute("git fetch origin #{GIT_MAIN_FETCH_REFS}", exit_on_error: true)
+  end
+end
+
+def create_repo(repo)
+  FileUtils.mkdir_p(cloned_repo_dir(repo))
+  puts "**** Creating #{cloned_repo_dir(repo)}"
+  within_cloned_repo_dir(repo) do
+    ErrorEmittingExecutor.execute("git clone --depth=5 git@github.com:#{repo}.git .", exit_on_error: true)
+    ErrorEmittingExecutor.execute('bundle install', exit_on_error: true)
+  end
+end
+
+def within_cloned_repo_dir(repo, &block)
+  Dir.chdir(cloned_repo_dir(repo)) do
+    # Execute commands in the project-specific bundler context, rather
+    #   than access-update-scripts' bundler context.
+    Bundler.with_unbundled_env do
+      block.call
+    end
+  end
+end
+
+def cloned_repo_dir(repo)
+  File.join(WORK_DIR, repo)
+end
+
+## Actual Script Below
+
+begin
+  pr_list = create_prs(repos)
+  puts "\n********** Summary **********"
+
+  if pr_list.all? { |result| result[:pr_link].nil? || result[:pr_link].empty? }
+    puts 'No PRs were created.'
+    puts '********** End Summary **********'
+    exit
+  end
+
+  unless pr_list.all? { |result| !result[:pr_link].nil? && !result[:pr_link].empty? }
+    puts '*Some* PRs were not created: '
+    pr_list.map do |result|
+      if result[:pr_link].nil? || result[:pr_link].empty?
+        puts "#{result[:repo]} did NOT create a PR for updated cocina gems."
+        puts '********** End Summary **********'
+      end
+    end
+    exit
+  end
```

Comment:
I offered to refactor this a bit in the new year. Will test against a small number of repos like @ndushay suggests.